src/main/java/de/lukweb/justmail/socket/ImapSocketHandler.java

```diff
@@ -65,16 +65,16 @@ public void run() {
                 }
                 cache += (char) read;
                 if (!(cache.endsWith("\n\r") || cache.endsWith("\n"))) continue;
-                /* if (session.getCallback() != null) {
+                if (session.getCallback() != null) {
                     boolean breakCB = session.getCallback().callback(cache);
                     if (breakCB) session.setCallback(null);
                     cache = "";
                     continue;
-                } */
-                String[] split = cache.split(" ", 1);
```

Comment:
If you set limit to 1, the split Function returns an ArrayList with 1 String ([Split-Function](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#split(java.lang.String,%20int))). So my change is right to make it work